packages/router/src/page-loader.js

```diff
@@ -1,35 +1,68 @@
 import { useContext } from 'react'
 
-import { createNamedContext } from './internal'
+import { createNamedContext, ParamsContext } from './internal'
 
 export const PageLoadingContext = createNamedContext('PageLoading')
 
 export const usePageLoadingContext = () => useContext(PageLoadingContext)
 
-export class PageLoader extends React.PureComponent {
+export class PageLoader extends React.Component {
   state = {
     Page: undefined,
     pageName: undefined,
     slowModuleImport: false,
   }
 
+  propsChanged = (p1, p2) => {
+    if (p1.spec.name !== p2.spec.name) {
+      return true
+    }
+    if (JSON.stringify(p1.params) !== JSON.stringify(p2.params)) {
+      return true
+    }
+    return false
+  }
```

Comment:
I notice that React warns against using JSON.stringify for deep equality, I imagine because it's slow.

Maybe you could use: https://www.npmjs.com/package/lodash.isequal instead?

![image](https://user-images.githubusercontent.com/44849/95682146-0b010f80-0be4-11eb-93a9-05ffcfcd814b.png)

Reply:
+1, I also think its a tiny bit more readable.